src/visitor.rs

```diff
@@ -166,39 +166,29 @@ impl<'a, 'v> visit::Visitor<'v> for FmtVisitor<'a> {
-                        let multi_line_budget = self.config.ideal_width - block_indent;
-                        let formatted = self.rewrite_use_list(block_indent,
-                                                              one_line_budget,
-                                                              multi_line_budget,
-                                                              path,
-                                                              path_list,
-                                                              item.vis,
-                                                              item.span);
-
-                        if let Some(new_str) = formatted {
-                            self.format_missing_with_indent(item.span.lo);
-                            self.changes.push_str_span(item.span, &new_str);
-                        } else {
-                            // Format up to last newline
-                            let span = codemap::mk_sp(self.last_pos, item.span.lo);
-                            let span_end = match self.snippet(span).rfind('\n') {
-                                Some(offset) => self.last_pos + BytePos(offset as u32),
-                                None => item.span.lo
-                            };
-                            self.format_missing(span_end);
-                        }
-
+                let vis = utils::format_visibility(item.vis);
+                let offset = self.block_indent + vis.len() + "use ".len();
+                let context = RewriteContext {
+                    codemap: self.codemap, config: self.config, block_indent: self.block_indent };
+                // 1 = ";"
+                match vp.rewrite(&context, self.config.max_width - offset - 1, offset) {
+                    Some(ref s) if s.len() == 0 => {
+                        // Format up to last newline
+                        let span = codemap::mk_sp(self.last_pos, item.span.lo);
+                        let span_end = match self.snippet(span).rfind('\n') {
+                            Some(offset) => self.last_pos + BytePos(offset as u32),
+                            None => item.span.lo
+                        };
+                        self.format_missing(span_end);
                         self.last_pos = item.span.hi;
                     }
-                    ast::ViewPath_::ViewPathGlob(_) => {
+                    Some(ref s) => {
+                        let s = format!("{}use {};", vis, s);
                         self.format_missing_with_indent(item.span.lo);
-                        // FIXME convert to list?
+                        self.changes.push_str_span(item.span, &s);
+                        self.last_pos = item.span.hi;
                     }
-                    ast::ViewPath_::ViewPathSimple(_,_) => {
+                    None => {
                         self.format_missing_with_indent(item.span.lo);
```

Comment:
If we can't reformat, we should probably use item.span.hi here so we get all the text we can't handle (in theory, this shouldn't make any difference, it should be caught later, but I think this is more logical).